File: SOFTWARE ENGINNER TESTS/test-03.ts

function isAlpha(input: string): boolean {
    const regex = /^[A-Za-z]+$/;
    return regex.test(input);
}


function isAlphabeticPalindrome(code: string): number {
    // Write your code here
   let arrayOfAlpha: string[] = [];
    for (let i = 0; i < code.length; i ++) {
      if (isAlpha(code[i]))
         arrayOfAlpha[i] = code[i];
    }

    if (arrayOfAlpha.length % 2 != 0)
      return 0;
    for (let i = 0; i < code.length /2; i ++) {
      if(code[i] != code[code.length - 1 - i])
         return 0;
    }
    return 1;
}

console.log(isAlphabeticPalindrome("abc123cba"));
 

File: SOFTWARE ENGINNER TESTS/test-03.py
def isAlphabeticPalindrome(code):
    # Write your code here
    alpha_array = []
    for letter in code:
        if ord(letter) < 33 or ord(letter) > 126:
            return 0
        if 'a' <= letter <= 'z' or 'A' <= letter <= 'Z':
            alpha_array.append(letter.lower())
    # [start:stop:step] adds a step value, which tells Python how to move through the list.
    return int(alpha_array == alpha_array[::-1])
    

if __name__ == '__main__':

    print(isAlphabeticPalindrome(" "))

    